janus_core/helpers/utils.py

```diff
@@ -191,23 +163,21 @@ def spacegroup(
     )
 
 
-def none_to_dict(dictionaries: list[Optional[dict]]) -> list[dict]:
+def none_to_dict(dictionaries: Sequence[Optional[dict]]) -> Generator[dict]:
     """
     Ensure dictionaries that may be None are dictionaires.
 
     Parameters
     ----------
-    dictionaries : list[dict]
-        List of dictionaries that be be None.
+    dictionaries : Sequence[dict]
+        Sequence of dictionaries that be be None.
 
     Returns
     -------
-    list[dict]
-        Dictionaries set to {} if previously None.
+    Generator[dict]
+        Generator of dictionaries set to {} if previously None.
```

Comment:
```rst
Yields
------
dict
    Input dictionary or ``{}`` if empty or `None`.

Reply:
Even though it's not from a `yield` statement?

Reply:
Hm... good point. I'm not sure... Maybe we should just change it to a yield from? I don't know.

Reply:
> Even though it's not from a `yield` statement?

numpydoc does indeed complain about no Returns section, so yep changed to `yield from`